design/cryptography.md

```diff
@@ -54,6 +55,23 @@ An additional complexity is that Obscuro has the requirement to support temporar
   - Transactions without Revelation Period specified are encrypted by default using the smallest unit of time revelation period
 
 
+### 4. Reveal the rollup transactions
+- L1 blocks are used as a clock mechanism
+  - There are 5 revelation periods
+    * XS - 12 seconds (1 block)
+    * S - 1 hour (3600/12 = 300 blocks)
+    * M - 1 day (24 * 3600/12 = 7200 blocks)
+    * L - 1 month (30 * 24 * 3600/12 = 216000 blocks)
+    * XL - 1 year (12 * 30 * 24 * 3600/12 = 2592000 blocks)
+- Central Sequencer stores the rollup revelation keys in a database with the corresponding decrypt time.
```

Comment:
what are the requirements?

Reply:
The requirements in the white paper are l1 blocks as the clock, the 5 revelation periods, and the key independent release. These are described in the section.

Maybe I can make them clearer. I've rewritten the section. But I'm not sure I understand if that's what you're asking.